admin/init.sls

```diff
@@ -27,9 +27,36 @@ UTC:
     - mode: 644
     - source: salt://{{ tpldir }}/files/hosts
 
-{% for ssh_user in admin.ssh_users %}
-sshkey-{{ ssh_user }}:
-  ssh_auth.present:
+# FIXME we should also explicitly AllowUsers usera userb userc
+/etc/sshd_config:
+  file.managed:
     - user: root
-    - source: salt://{{ tpldir }}/files/ssh/{{ ssh_user }}.pub
+    {% if grains['os'] == 'MacOS' %}
+    - group: wheel
+    {% elif grains['os'] == 'Ubuntu' %}
+    - group: root
+    {% endif %}
+    - mode: 644
+    - source: salt://{{ tpldir }}/files/sshd_config
+
+
+{% for ssh_user in admin.ssh_users %}
+{{ ssh_user }}:
+    user.present:
```

Comment:
Before `user.present`, we should also use `group.present` to create a group for each user with the same name. We should require the group state from the user state and set `gid_from_name: True` in the user state.